core/server/services/bulk-email/index.js

```diff
@@ -46,24 +47,45 @@ module.exports = {
      * @param {Email} message - The message to send
      * @param {[EmailAddress]} recipients - the recipients to send the email to
      * @param {[object]} recipientData - list of data keyed by email to inject into the email
-     * @returns {Promise<boolean>} A promise representing the success of the email sending
+     * @returns {Array{Promise<boolean>}} An array of promises representing the success of the batch email sending
      */
     async send(message, recipients, recipientData) {
+        let BATCH_SIZE = 1000;
```

Comment:
I didn't find any mention of the limit in the official docs. The best I could get was this reference - mailgun/mailgun-php#469